ui-core/src/components/inputs/checkbox/type.ts

```diff
@@ -0,0 +1,24 @@
+import { CheckboxProps } from './Checkbox';
+
+export type checkboxListItem = {
+  id: number;
+  parentId?: number; // Make parentId optional for root items
+  checkboxProps: CheckboxProps;
```

Comment:
I think props is a better name
```suggestion
  props: CheckboxProps;
```